src/algebra/lie/nilpotent.lean

```diff
@@ -67,6 +78,32 @@ class is_nilpotent : Prop :=
 instance trivial_is_nilpotent [is_trivial L M] : is_nilpotent R L M :=
 ⟨by { use 1, change ⁅⊤, ⊤⁆ = ⊥, simp, }⟩
 
+lemma nilpotent_endo_of_nilpotent_module [hM : is_nilpotent R L M] :
+  ∃ (k : ℕ), ∀ (x : L), (to_endomorphism R L M x)^k = 0 :=
```

Comment:
I guess we really need some sort of `is_nilpotent` predicate for endofunctions/endomorphisms. But maybe not in this PR.

Reply:
Yes. I briefly considered introducing it but also decided it should probably be separate, especially as the lemma here says that the same `k` works for all the endomorphisms.

I.e., the obvious `is_nilpotent` predicate would turn my `∃ (k : ℕ), ∀ (x : L)` into the weaker `∀ (x : L), ∃ (k : ℕ)`. Admittedly the weaker version is all I need but who knows about the future?